tpl/Dockerfile.almalinux

```diff
@@ -0,0 +1,372 @@
+# This library is considered production ready.
+ARG VER_LUA_RESTY_LRUCACHE=0.11
+ENV VER_LUA_RESTY_LRUCACHE=$VER_LUA_RESTY_LRUCACHE
+
+# headers-more-nginx-module
+# https://github.com/openresty/headers-more-nginx-module/commits/master
+ARG VER_OPENRESTY_HEADERS=a4a0686605161a6777d7d612d5aef79b9e7c13e0
+ENV VER_OPENRESTY_HEADERS=$VER_OPENRESTY_HEADERS
+
+# lua-resty-cookie
+# https://github.com/cloudflare/lua-resty-cookie/commits/master
+ARG VER_CLOUDFLARE_COOKIE=99be1005e38ce19ace54515272a2be1b9fdc5da2
+ENV VER_CLOUDFLARE_COOKIE=$VER_CLOUDFLARE_COOKIE
+
+# lua-resty-dns
+# https://github.com/openresty/lua-resty-dns/tags
+ARG VER_OPENRESTY_DNS=0.22
+ENV VER_OPENRESTY_DNS=$VER_OPENRESTY_DNS
+
+# lua-resty-memcached
+# https://github.com/openresty/lua-resty-memcached/tags
+ARG VER_OPENRESTY_MEMCACHED=0.16
+ENV VER_OPENRESTY_MEMCACHED=$VER_OPENRESTY_MEMCACHED
+
+# lua-resty-mysql
+# https://github.com/openresty/lua-resty-mysql/tags
+ARG VER_OPENRESTY_MYSQL=0.24
+ENV VER_OPENRESTY_MYSQL=$VER_OPENRESTY_MYSQL
+
+# lua-resty-redis
+# https://github.com/openresty/lua-resty-redis/releases
+ARG VER_OPENRESTY_REDIS=0.29
+ENV VER_OPENRESTY_REDIS=$VER_OPENRESTY_REDIS
+
+# lua-resty-shell
+# https://github.com/openresty/lua-resty-shell/tags
+ARG VER_OPENRESTY_SHELL=0.03
+ENV VER_OPENRESTY_SHELL=$VER_OPENRESTY_SHELL
+
+# lua-resty-signal
+# https://github.com/openresty/lua-resty-signal/releases
+ARG VER_OPENRESTY_SIGNAL=0.03
+ENV VER_OPENRESTY_SIGNAL=$VER_OPENRESTY_SIGNAL
+
+# lua-tablepool
+# https://github.com/openresty/lua-tablepool/releases
+ARG VER_OPENRESTY_TABLEPOOL=0.02
+ENV VER_OPENRESTY_TABLEPOOL=$VER_OPENRESTY_TABLEPOOL
+
+# lua-resty-upstream-healthcheck
+# https://github.com/openresty/lua-resty-upstream-healthcheck/tags
+ARG VER_OPENRESTY_HEALTHCHECK=0.06
+ENV VER_OPENRESTY_HEALTHCHECK=$VER_OPENRESTY_HEALTHCHECK
+
+# lua-resty-websocket
+# https://github.com/openresty/lua-resty-websocket/tags
+ARG VER_OPENRESTY_WEBSOCKET=0.08
+ENV VER_OPENRESTY_WEBSOCKET=$VER_OPENRESTY_WEBSOCKET
+
+# lua-rocks
+# https://luarocks.github.io/luarocks/releases/
+ARG VER_LUAROCKS=3.8.0
+ENV VER_LUAROCKS=$VER_LUAROCKS
+
+# lua-upstream-nginx-module
+# https://github.com/openresty/lua-upstream-nginx-module/tags
+ARG VER_LUA_UPSTREAM=0.07
+ENV VER_LUA_UPSTREAM=$VER_LUA_UPSTREAM
+
+# nginx-lua-prometheus
+# https://github.com/knyar/nginx-lua-prometheus/tags
+ARG VER_PROMETHEUS=0.20210206
+ENV VER_PROMETHEUS=$VER_PROMETHEUS
+
+# stream-lua-nginx-module
+# https://github.com/openresty/stream-lua-nginx-module/tags
+ARG VER_OPENRESTY_STREAMLUA=0.0.10
+ENV VER_OPENRESTY_STREAMLUA=$VER_OPENRESTY_STREAMLUA
+
+# https://github.com/nginx/nginx/releases
+ARG VER_NGINX={{VER_NGINX}}
+ENV VER_NGINX=$VER_NGINX
+ARG NGX_CFLAGS="-g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fPIC"
+ENV NGX_CFLAGS=$NGX_CFLAGS
+ARG NGX_LDOPT="-Wl,-rpath,/usr/local/lib -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -pie -lpcre"
```

Comment:
added flag `-lpcre` as workaround as per https://github.com/openresty/lua-nginx-module/issues/1984#issuecomment-1003709530, mostly due to [nginx-lua-module not supporting pcre2](https://github.com/openresty/lua-nginx-module/issues/1984)